packages/react-components/react-color-picker-preview/library/src/components/ColorArea/useColorArea.ts

```diff
@@ -9,23 +18,143 @@ import type { ColorAreaProps, ColorAreaState } from './ColorArea.types';
+    event.preventDefault();
+
+    requestColorChange(event.nativeEvent);
+
+    targetDocument?.addEventListener('mousemove', handleDocumentMouseMove);
+    targetDocument?.addEventListener('mouseup', handleDocumentMouseUp, { once: true });
+  });
+
+  const _onChange: React.ChangeEventHandler<HTMLInputElement> = useEventCallback(event => {
+    const newColor = { h: color.h, s: Number(xRef.current!.value) / 100, v: Number(yRef.current?.value) / 100, a: 1 };
+
+    setColor(newColor);
+    onChange?.(event, {
+      type: 'change',
+      event,
+      color: newColor,
+    });
+  });
+
+  const _onKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'ArrowUp' || event.key === 'ArrowDown') {
+      event.preventDefault();
+
+      const newY = event.key === 'ArrowUp' ? clamp(value + 1, MIN, MAX) : clamp(value - 1, MIN, MAX);
+      const newColor: HsvColor = {
+        ...color,
+        s: color.s,
+        v: newY / 100,
+      };
+      setColor(newColor);
+      onChange?.(event, { type: 'change', event, color: newColor });
+    }
+  });
+
+  const rootVariables = {
+    [colorAreaCSSVars.areaXProgressVar]: `${saturation}%`,
+    [colorAreaCSSVars.areaYProgressVar]: `${value}%`,
+    [colorAreaCSSVars.thumbColorVar]: 'transparent',
+    [colorAreaCSSVars.mainColorVar]: `hsl(${color.h}, 100%, 50%)`,
+  };
+
+  const state: ColorAreaState = {
     components: {
-      // TODO add each slot's element type or component
+      inputX: 'input',
+      inputY: 'input',
       root: 'div',
+      thumb: 'div',
     },
-    // TODO add appropriate slots, for example:
-    // mySlot: resolveShorthand(props.mySlot),
-    root: slot.always(
-      getIntrinsicElementProps('div', {
-        ref,
-        ...props,
-      }),
-      { elementType: 'div' },
-    ),
+    root: slot.always(getIntrinsicElementProps('div', { ...rest, ref }), { elementType: 'div' }),
+    inputX: slot.always(inputX, {
+      defaultProps: {
+        id: useId('sliderX-', props.id),
+        type: 'range',
+        ref: xRef,
+      },
+      elementType: 'input',
+    }),
+    inputY: slot.always(inputY, {
+      defaultProps: {
+        id: useId('sliderY-', props.id),
+        type: 'range',
+        ref: yRef,
+      },
+      elementType: 'input',
+    }),
+    thumb: slot.always(thumb, { elementType: 'div' }),
+  };
+
+  state.root.ref = useMergedRefs(state.root.ref, rootRef);
+  state.thumb.ref = useMergedRefs(state.thumb.ref, focusWithinRef);
+
+  state.root.style = {
+    ...rootVariables,
+    ...state.root.style,
   };
+
+  state.root.onMouseDown = useEventCallback(mergeCallbacks(state.root.onMouseDown, _onMouseDown));
+  state.inputX.onChange = useEventCallback(mergeCallbacks(state.inputX.onChange, _onChange));
+  state.inputY.onChange = useEventCallback(mergeCallbacks(state.inputY.onChange, _onChange));
+  state.inputX.onKeyDown = useEventCallback(mergeCallbacks(state.inputX.onKeyDown, _onKeyDown));
```

Comment:
Does `inputX` gets focus? I.e. is it a right place for mousedown handler?

Reply:
Input gets focus only on keyboard interaction because thumb use "focus within".